pkg/controller/external_tfpluginsdk.go

```diff
@@ -216,16 +216,8 @@ func (c *TerraformPluginSDKConnector) applyStateFuncToParam(sc *schema.Schema, p
 				param = hclProccessedParam
 			}
 		}
-		if sc.StateFunc != nil {
-			return sc.StateFunc(param)
-		}
-		return param
-	case schema.TypeBool, schema.TypeInt, schema.TypeFloat:
-		if sc.StateFunc != nil {
-			return sc.StateFunc(param)
-		}
 		return param
-	case schema.TypeInvalid:
+	case schema.TypeBool, schema.TypeInt, schema.TypeFloat, schema.TypeInvalid:
 		return param
 	default:
 		return param
```

Comment:
We can merge this case with the `default` case:
```suggestion
	default:
		return param
```

Reply:
Done!